addons/binding/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/discovery/BigAssFanDiscoveryService.java

```diff
@@ -0,0 +1,256 @@
+            device.setLabel(matcher.group(1));
+            device.setMacAddress(matcher.group(2));
+            String[] modelParts = matcher.group(3).split(",");
+            if (modelParts.length != 2) {
+                logger.info("Unable to extract device type from discovery message");
+                return;
+            }
+            device.setType(modelParts[0]);
+            device.setModel(modelParts[1]);
+
+            deviceDiscovered(device);
+        }
+    }
+
+    private synchronized void deviceDiscovered(BigAssFanDevice device) {
+        logger.info("Device discovered: {}", device);
+
+        if (device.isSwitch()) {
+            logger.debug("Switches (controllers) are not handled currently");
+            return;
+        } else if (device.isFan()) {
+            logger.debug("Add fan with IP={}, MAC={}", device.getIpAddress(), device.getMacAddress());
+        } else {
+            logger.debug("Discovered unknown device type {} at IP={}", device.getModel(), device.getIpAddress());
+            return;
+        }
+
+        // We got a valid discovery message. Process it as a potential new thing
+        String serialNumber = device.getMacAddress().replace(":", "");
+
+        Map<String, Object> properties = new HashMap<>();
+        properties.put(THING_PROPERTY_MAC, device.getMacAddress());
+        properties.put(THING_PROPERTY_IP, device.getIpAddress());
+        properties.put(THING_PROPERTY_LABEL, device.getLabel());
+        properties.put(Thing.PROPERTY_SERIAL_NUMBER, serialNumber);
+        properties.put(Thing.PROPERTY_MODEL_ID, device.getModel());
+        properties.put(Thing.PROPERTY_VENDOR, "Haiku");
+
+        ThingUID uid = new ThingUID(THING_TYPE_FAN, serialNumber);
+        // If there's not a thing and it's not in the inbox, create the discovery result
+        if (callback != null && callback.getExistingDiscoveryResult(uid) == null
+                && callback.getExistingThing(uid) == null) {
+
+            logger.info("Creating discovery result for UID={}, IP={}", uid, device.getIpAddress());
+            DiscoveryResult result = DiscoveryResultBuilder.create(uid).withProperties(properties)
+                    .withLabel(device.getLabel()).build();
+
+            thingDiscovered(result);
+        } else {
+            logger.debug("Thing or inbox entry already exists for UID={}, IP={}", uid, device.getIpAddress());
+        }
+    }
+
+    private void schedulePollJob() {
+        logger.debug("Scheduling poll job to run every {} seconds starting in {} sec", POLL_FREQ, POLL_DELAY);
+        cancelPollJob();
+        pollJob = scheduler.scheduleAtFixedRate(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    discoveryListener.pollForDevices();
+                } catch (RuntimeException e) {
+                    logger.error("Poll job got unexpected exception: {}", e.getMessage(), e);
```

Comment:
Can you make this a warning errors are only for things threatening the stability of the complete openHAB instance.